components/Shipping/basic_usage.rst

```diff
@@ -0,0 +1,666 @@
+Basic Usage
+===========
+
+Shipping Category
+-----------------
+
+Every shipping category has two identifiers, an ID and name. You can access those by calling ``->getId()`` and ``->getName()``
+methods respectively. The name is mutable, so you can change it by calling ``->setName('Regular')`` on the shipping category instance.
+
+Shipping Method
+---------------
+
+Every shipping method has two identifiers, an ID and name. You can access those by calling ``->getId()`` and ``->getName()``
+methods respectively. The name is mutable, so you can change it by calling ``->setName('FedEx')`` on the shipping method instance.
+
+Setting Shipping Category
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Every shipping method can have shipping category. You can simply set or unset it by calling ``->setCategory()``. The category
+can have optional requirements which are set by ``->setCategoryRequirement()`` method.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+        use Sylius\Component\Shipping\Model\ShippingCategory;
+        use Sylius\Component\Shipping\Model\ShippingMethodInterface;
+
+        $shippingCategory = new ShippingCategory();
+        $shippingCategory->setName('Regular'); // Regular weight items
+
+
+        $shippingMethod = new ShippingMethod();
+        $shippingMethod->setCategory($shippingCategory); //default null, detach
+        $shippingMethod->getCategory(); // Output will be ShippingCategory object
+        $shippingMethod->setCategory();
+        $shippingMethod->getCategory(); // Output will be null
+        $shippingMethod
+        ->setCategoryRequirement(ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_ALL);
+        // the shipping method is now support any shipment (or shipping subject) which contains
+        // at least one shippable with the same category. If shipment have many shippable object,
+        // all of them have to have the same category.
+        $shippingMethod
+        ->setCategoryRequirement(ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_NONE);
+        // if none shippables (in shipment) have reference the same category as shippingMethod,
+        // the shippingMethod is allowed for this shipment
+        $shippingMethod
+        ->setCategoryRequirement(ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_ANY);
+        // if one  of the shippables in shipping have the same category as shippingMethod,
+        // the shipping Method is allowed for this shipment
```

Comment:
I don't think we need to be that specific. Let's just show that you can attach and detach a category to the method. Also, when detaching - be explicit - setCategory(null).